crates/oxc_estree/src/lib.rs

```diff
@@ -1,3 +1,25 @@
+#[cfg(feature = "serialize")]
+use serde::ser::{Serialize, SerializeSeq, Serializer};
+
 /// Empty trait that will be used later for custom serialization and TypeScript
 /// generation for AST nodes.
 pub trait ESTree {}
+
+#[cfg(feature = "serialize")]
+pub struct AppendTo<'a, TVec, TChild>(pub &'a [TVec], pub &'a Option<TChild>);
```

Comment:
Personally, I generally prefer named fields over tuple structs. It's more verbose, but it makes it easier to read and understand the code.

`AppendTo` could also do with a comment explaining what it does. Oxc's codebase is not all as well-commented as it should be, but we're trying to improve in that department.